src/user/user.service.ts

```diff
@@ -1,298 +1,254 @@
+                throw new ConflictException('Username is already taken.');
+            else throw new InternalServerErrorException('Update User failed');
+        }
     }
 
-    const resetCode = crypto.randomBytes(64).toString('hex');
-    const resetObject = new this.resetSchema({
-      userId: user._id,
-      verificationCode: resetCode
-    })
-    const result = await resetObject.save()
+    async remove(id: ObjectId, requestingUser: JwtUserDto): Promise<User> {
+        if (
+            requestingUser.role !== Role.Admin &&
+            id !== requestingUser.userId
+        ) {
+            throw new UnauthorizedException();
+        }
 
-    await this.mailService.sendPasswordReset(user.username, user.email, resetCode)
+        const user = await this.userSchema.findByIdAndDelete(id);
 
-  }
+        if (!user) throw new NotFoundException();
 
-  async findReset(userId: ObjectId): Promise<EmailVerify> {
-    const verifyObject = await this.resetSchema.findOne({
-      'userId': userId
-    }).lean()
-    if (!verifyObject) {
-      throw new NotFoundException()
-    }
-    return verifyObject
-  }
-
-  /**
-   * Overwrites the password with the new one
-   * @param code - resetcode as passed in url
-   * @param password new password
-   * @returns 
-   */
-  async validatePasswordReset(code: string, password: string) {
-    const resetObject = await this.resetSchema.findOneAndDelete({
-      'verificationCode': code
-    })
-
-    if (!resetObject) {
-      throw new NotFoundException()
+        return user;
     }
 
-    if (Date.now() - resetObject._id.getTimestamp() > +process.env.RESET_TTL) {
-      throw new BadRequestException('Token expired.')
+    async validateVerifyCode(userId: ObjectId): Promise<boolean> {
+        let user: User;
+        try {
+            user = await this.findOneById(userId);
+        } catch (error) {
+            // This is necessary as a not found exception would overwrite the guard response
+            return false;
+        }
+        if (!user) return false; // This should never happen but just in case
+        if (user.status !== UserStatus.UNVERIFIED) {
+            return false;
+        }
+        return true;
     }
 
-    const hash = await bcyrpt.hash(password, 12)
+    async veryfiyUser(userId: ObjectId): Promise<User> {
+        const user = await this.userSchema.findByIdAndUpdate(userId, {
+            status: UserStatus.ACTIVE
+        });
 
-    const updatedUser: User = await this.userSchema.findByIdAndUpdate(resetObject.userId, {
-        password: hash
-      }, {
-        new: true
-    })
+        //failsave that should never occur so istanbul ignore\
+        /* istanbul ignore next */
+        if (!user) {
+            throw new NotFoundException();
+        }
 
-    if (!updatedUser) {
-      throw new NotFoundException()
+        return user;
     }
 
-    return updatedUser
-  }
+    async transformToReturn(user: User): Promise<returnUser> {
```

Comment:
Maybe strip the password directly at mongoose

Reply:
possible but i would like to keep it this way.